src/storage/config.rs

```diff
@@ -333,6 +334,14 @@ impl IORateLimitConfig {
             );
             self.other_priority = IOPriority::High;
         }
+        if self.gc_priority != self.foreground_write_priority {
+            warn!(
+                "GC writes are merged with foreground writes. To avoid priority inversion, change \
+                  priority for IOType::Gc from {:?} to {:?}",
+                self.gc_priority, self.foreground_write_priority,
+            );
+            self.gc_priority = self.foreground_write_priority;
```

Comment:
When compaction filter is enabled, does it affect compaction IO priority?

Reply:
No, the `Gc` type guard is only scoped around actual writes without contaminating compaction thread.